server/accept.go

```diff
@@ -0,0 +1,134 @@
+package server
```

Comment:
Maybe existing battle-tested code for this would have been easier to add: https://github.com/timewasted/go-accept-headers/blob/master/accept.go